liota/dccs/iotcc.py

```diff
@@ -0,0 +1,248 @@
+import Queue
+
+from liota.dccs.dcc import DataCenterComponent
+from liota.lib.protocols.helix_protocol import HelixProtocol
+from liota.entities.metrics.metric import Metric
+from liota.utilities.utility import getUTCmillis, LiotaConfigPath
+from liota.utilities.si_unit import parse_unit
+from liota.entities.registered_entity import RegisteredEntity
+
+log = logging.getLogger(__name__)
+
+
+class IotControlCenter(DataCenterComponent):
+    """ The implementation of vROPS cloud provider solution
+
+    """
+
+    def __init__(self, username, password, con):
+        log.info("Logging into DCC")
+        self.con = con
+        self.username = username
+        self.password = password
+        self.proto = HelixProtocol(self.con, username, password)
+
+        def on_receive_safe(msg):
+            try:
+                log.debug("Received msg: {0}".format(msg))
+                json_msg = json.loads(msg)
+                self.proto.on_receive(json.loads(msg))
+                log.debug("Processed msg: {0}".format(json_msg["type"]))
+                if json_msg["type"] == "connection_verified":
+                    log.info("Verified")
+                    exit()
+            except Exception:
+                raise
+                log.exception(
+                    "Error received on connecting to DCC instance. Please verify the credentials and try again.")
+
+        thread = threading.Thread(target=self.con.run)
+        self.con.on_receive = on_receive_safe
+        thread.daemon = True
+        thread.start()
+        thread.join()
+        log.info("Logged in to DCC successfully")
+
+    def register(self, entity_obj):
+        """ Register the objects
+
+        """
+        if isinstance(entity_obj, Metric):
+            self.publish_unit(entity_obj.parent, entity_obj.name, entity_obj.unit)
+            return entity_obj.register(self, entity_obj.parent.reg_entity_id)
+        else:
+            if not hasattr(entity_obj, "reg_entity_id"):
+                log.info("Creating resource")
+            log.info("Resource Name: {0}".format(entity_obj.name))
+
+            def on_receive_safe(msg):
+                try:
+                    log.debug("Received msg: {0}".format(msg))
+                    if msg != "":
+                        json_msg = json.loads(msg)
+                        self.proto.on_receive(json.loads(msg))
+                        log.debug("Processed msg: {0}".format(json_msg["type"]))
+                        if json_msg["type"] == "create_or_find_resource_response":
+                            if json_msg["body"]["uuid"] != "null":
+                                log.info("FOUND RESOURCE: {0}".format(json_msg["body"]["uuid"]))
+                                self.reg_entity_id = json_msg["body"]["uuid"]
+                                exit()
+                            else:
+                                log.info("Waiting for resource creation")
+                                time.sleep(5)
```

Comment:
It is a wait call to check back if the resource is created or not. I can remove this time.sleep but then the too many request will fire up from the liota to IoTCC.